.travis.yml

```diff
@@ -1,47 +1,38 @@
-language: python
+language: generic
 sudo: false
-python:
-  - 2.7_with_system_site_packages
-  - 3.4
-addons:
-  apt:
-    packages:
-    - python-qt4
-    - python-qt4-dev
-    - python-qt4-gl
-    - python-pip
-    - python-numpy
-    - python-wxtools
-    - ccache
-    - cmake
-env:
-  - ETS_TOOLKIT=qt4
-  - ETS_TOOLKIT=null
-  - ETS_TOOLKIT=wx
 matrix:
+  include:
+    - env: RUNTIME=2.7 TOOLKIT=null
+    - env: RUNTIME=2.7 TOOLKIT=pyqt
+    - env: RUNTIME=2.7 TOOLKIT=pyside
+    - env: RUNTIME=2.7 TOOLKIT=wx
+    - env: RUNTIME=3.5 TOOLKIT=null
+    - env: RUNTIME=3.5 TOOLKIT=pyqt
+    - os: osx
+      env: RUNTIME=2.7 TOOLKIT=null
+    - os: osx
+      env: RUNTIME=2.7 TOOLKIT=pyqt
+    - os: osx
+      env: RUNTIME=2.7 TOOLKIT=pyside
+    - os: osx
+      env: RUNTIME=2.7 TOOLKIT=wx
+    - os: osx
+      env: RUNTIME=3.5 TOOLKIT=null
+    - os: osx
+      env: RUNTIME=3.5 TOOLKIT=pyqt
   allow_failures:
-    - env: ETS_TOOLKIT=wx
+    - env: RUNTIME=2.7 TOOLKIT=wx
+  fast_finish: true
 cache:
   directories:
     - $HOME/.cache
-    - $HOME/.ccache
```

Comment:
we should keep `ccache` since it is possible that some of the source despondencies might need a c compiler (e.g. cython code).